ipatests/test_integration/test_multidomain_ipa.py

```diff
@@ -0,0 +1,26 @@
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import MultiDomainIntegrationTest
+
+
+class TestMultidomain(MultiDomainIntegrationTest):
+    num_clients = 1
+    num_replicas = 1
+    num_trusted_clients = 1
+    num_trusted_replicas = 1
+    topology = 'line'
+
+    def test_multidomain_setup(self):
+        """
+        Test services on multidomain topology.
+        """
+
+        for host in (self.master, self.replicas[0],
+                     self.trusted_master, self.trusted_replicas[0]
+                     ):
+            tasks.start_ipa_server(host)
+
+        for host in (self.master, self.replicas[0],
+                     self.trusted_master, self.trusted_replicas[0],
+                     self.clients[0], self.trusted_clients[0]
+                     ):
+            tasks.kinit_admin(host)
```

Comment:
Before calling trust-add, the test also needs to configure forward zones:
```

        # Add DNS forwarder to trusted domain on ipa domain
        self.master.run_command([
            "ipa", "dnsforwardzone-add", self.trusted_master.domain.name,
            "--forwarder", self.trusted_master.ip,
            "--forward-policy=only"
        ])
        self.trusted_master.run_command([
            "ipa", "dnsforwardzone-add", self.master.domain.name,
            "--forwarder", self.master.ip,
            "--forward-policy=only"
        ])
```

The following part will not work yet (unless you use the copr abbra/wip-ipa-trust, but you can already add it and comment it out:
```

        # Establish trust
        self.master.run_command([
            "ipa", "trust-add", "--type=ipa",
            "--admin", "admin@{}".format(self.trusted_master.domain.realm),
            "--range-type=ipa-ad-trust-posix",
            "--password", "--two-way=true",
            self.trusted_master.domain.name
        ], stdin_text=self.trusted_master.config.admin_password)
```
You can also prepare code for the ranges.